java/src/jmri/jmrit/dispatcher/ActiveTrain.java

```diff
@@ -1002,6 +1017,33 @@ protected AllocatedSection reverseAllAllocatedSections() {
         }
         return aSec;
     }
+    // Reverse a specfic section
+    // as the syncronzed cannot lock the array we try twice in case
+    // of concurrent updates.
+    protected AllocatedSection DONTUSEreverseOneAllocatedSections(String sectionToReverse) {
+        AllocatedSection aSec = null;
+        int iCount = 0;
+        boolean bResetDone = false;
+        while (!bResetDone && iCount < 10) {
+            log.info("Try reset[" + sectionToReverse + "]");
+            for (int i = 0; i < mAllocatedSections.size(); i++) {
+                aSec = mAllocatedSections.get(i);
+                if ( sectionToReverse.equals(aSec.getSectionName() ) ) {
+                    int dir = mTransit.getDirectionFromSectionAndSeq(aSec.getSection(), aSec.getSequence());
+                    if (dir == jmri.Section.FORWARD) {
+                        aSec.getSection().setState(jmri.Section.REVERSE);
+                    } else {
+                        aSec.getSection().setState(jmri.Section.FORWARD);
+                    }
+                    bResetDone = true;
+                    aSec.setStoppingSensors();
+                }
+            }
+            try { Thread.sleep(200); } catch (Exception e) {};
```

Comment:
This ```Exception``` needs to be an ```InterruptedException```. Changing this will allow the CI servers to pass.

Reply:
Will do